code/modules/food_and_drinks/recipes/tablecraft/recipes_pastry.dm

```diff
@@ -499,9 +499,18 @@
 	parts = list(
 		/obj/item/paper = 1
 	)
+	/// The paper to put in the cookie
+	var/obj/item/paper/fortune
 	result = /obj/item/food/fortunecookie
 	subcategory = CAT_PASTRY
 
+/datum/crafting_recipe/food/fortunecookie/check_requirements(mob/user, list/collected_requirements)
+	fortune = collected_requirements[/obj/item/paper][1]
+	return TRUE
+
+/datum/crafting_recipe/food/fortunecookie/on_craft_completion(mob/user, obj/item/food/fortunecookie/result)
+	result.fortune = fortune
```

Comment:
Tossing around a reference to a paper like this isn't really sane. Crafting recipes are singletons, there's only ever one at any one time, they shouldn't store info like this.

The paper is already "inside" the created fortune cookie regardless, because of the parts list. We just don't have a direct reference to it.

Reply:
removed